kong/router/atc.lua

```diff
@@ -96,12 +96,10 @@ end
 
 
 local function escape_str(str)
-  if str:find([[\]], 1, true) then
-    str = str:gsub([[\]], [[\\]])
-  end
-
-  if str:find([["]], 1, true) then
-    str = str:gsub([["]], [[\"]])
+  if str:find([[\]], 1, true) or
```

Comment:
Is this logic necessary? For use cases like regex, we can always use raw string even if escaping is not needed. This function can simply be named `raw_str(str)` instead.

Reply:
done.